### PR TITLE
Add Imply to PDDL domain parsing

### PR DESCRIPTION
**Pull request by sea-bass**
I was running PlanSys2 with a PDDL domain that had an `imply` in it, and noticed we did not support it.

So, now we do! Thankfully it was basically copypasta from the `forall` implementation that was already there... and it works :)

Closes https://github.com/PlanSys2/ros2_planning_system/issues/169

### CODE REVIEW

File: plansys2_pddl_parser/include/plansys2_pddl_parser/Domain.h [hunk 1]
```diff
@@ -313,6 +314,19 @@ class Domain {
...
+	// 		exit(1);
+	// 	}
+
+	// 	f.next();
+	// 	Imply * d = new Imply;
+	// 	d->parse( f, types[0]->constants, *this );
+
+	// 	if ( DOMAIN_DEBUG ) std::cout << d << "\n";
+	// 	derived.insert( d );
+	// }
```
Comment (sea-bass) [inline plansys2_pddl_parser/include/plansys2_pddl_parser/Domain.h:329]:
```suggestion
```

**Comment by fmrico** [comment]
Hi @sea-bass 

Could you add a simple test to verify that it is correctly parsed? Tests in this package are simple.

Thanks!!

**Comment by sea-bass** [comment]
> Hi @sea-bass
> 
> Could you add a simple test to verify that it is correctly parsed? Tests in this package are simple.
> 
> Thanks!!

Done! Added both `forall` (which wasn't there) and `imply` to the parser test domain. Let me know if you had something else in mind.

**Comment by fmrico** [comment]
Thanks!!! Merging :rocket:

**Comment by eshan-savla** [comment]
Hi @sea-bass,

First of all, thank you so much for your contribution! I don't mean to undermine your efforts with this, but I was going over your changes as a part of this issue #344 and something stood out to me. 

```C++
void Imply::parse( Stringreader & f, TokenStruct< std::string > & ts, Domain & d ) {
1.	f.next();
2.	f.assert_token( "(" );

3.	TokenStruct< std::string > fs = f.parseTypedList( true, d.types );
4.	params = d.convertTypes( fs.types );

5.	TokenStruct< std::string > fstruct( ts );
6.	fstruct.append( fs );

7.	f.next();
8.	f.assert_token( "(" );
9.	if ( f.getChar() != ')' ) {
10.		cond = d.createCondition( f );
11.		cond->parse( f, fstruct, d );
12.	}
13.	else ++f.c;

14.	f.next();
15.	f.assert_token( ")" );
}
```
I haven't been working with this repo and pddl for so long, so I might have a few things wrong, but I have a question. Lines 3 - 4 make sense in the case of _forall_, where the first set of tokens are always typed lists, however in the case of _imply_, the first set of tokens aren't typed lists, rather predicates. Considering this, would it not make more sense to somehow reuse lines 9-12? Also, the second predicate is being set as the condition, whereas in truth, shouldn't it be an evaluation of both predicates of imply or just the first one? Maybe saving both as two seperate conditions?

I might also be completely off base in my line of thought, so perhaps someone else could also weigh in and let me know if my thinking is on the right track? @fmrico, if you have the time, I would greatly value your thoughts on this.

**Pull request merged**